boards: Add lvgl pointer support on m5stack_core2

MrMarteng opened the pull request:
Added lvgl-pointer definition in devicetree, to enable touch-support in lvgl applications.
Has been tested with latest lvgl hello world sample.

Commit:
boards: Add lvgl pointer support on m5stack_core2

Added lvgl-pointer definition in devicetree, to enable
touch-support in lvgl applications.
Kscan support has been removed from m5stack_core2 as this was just a
temporary thing.
Has been tested with lvgl hello world sample.

Signed-off-by: Martin Kiepfer <m.kiepfer@teleschirm.org>

fabiobaltieri commented on boards/xtensa/m5stack_core2/m5stack_core2.dts:
```diff
@@ -154,7 +159,7 @@
 	};
 
-	ft5336@38 {
+	ft5336_touch: ft5336@38 {
```
Could you drop the `kscan-input` down below in the process? That was a temporary thing and can go now that we are pointing boards to the native driver.

MrMarteng replied:
I was expecting `kscan-input` is still needed for non-lvgl applications. Can you please explain in more detail what you mean with temporary thing? Is there an alternative for non-lvgl applications?

faxe1008 replied:
The kscan-input in the context of lvgl was used for compatibility. All touch drivers previously used the keyscan API. With the advent of the input subsystem touch drivers were converted to it. As a compatibility layer the kscan-input was added. Eventually we want to move away from using kscan nodes for lvgl pointer devices, since there is now the `lvgl_pointer_input` node.

fabiobaltieri replied:
Right, if the application still needs it then it can go in an overlay in the application itself.

faxe1008 commented on boards/xtensa/m5stack_core2/m5stack_core2.dts:
```diff
@@ -81,7 +86,7 @@
 &i2c0 {
 	status = "okay";
-	clock-frequency = <I2C_BITRATE_STANDARD>;
+	clock-frequency = <I2C_BITRATE_FAST>;
```
Is the bitrate change really needed here?

MrMarteng replied:
It was something I noticed during this change. All connected devices support 400kbit and I want to make sure high responsiveness. I created a separate commit for clarification.

faxe1008 commented on boards/xtensa/m5stack_core2/m5stack_core2_defconfig:
```diff
@@ -4,7 +4,7 @@ CONFIG_BOARD_M5STACK_CORE2=y
 
-CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_MAIN_STACK_SIZE=4096
 CONFIG_ESP_HEAP_MEM_POOL_REGION_1_SIZE=0
```
If the change of the main stack size is needed by the lvgl sample maybe adding a .conf in the sample instead would make sense? This increases the footprint also for non lvgl applications.

MrMarteng replied:
Thank you. You are absolutely right!
Added a config file on app level.

Commit:
boards: m5stack_core2: Increase i2c bitrate on i2c0

All periphals connected on i2c0 support 400kbit. The default speed has
been increased to improve responsiveness.

Signed-off-by: Martin Kiepfer <m.kiepfer@teleschirm.org>
